File: ohheck.help/Models/Data/Notification.cs
namespace ohheck.help.Models.Data {
    public class Notification : Common {
        public string foruser { get; set; }
        public Level level { get; set; }
        public string text { get; set; }
        public bool seen { get; set; }
        public virtual NotificationAction action { get; set; }
    }

    public enum Level {
        success,
        info,
        warning,
        error
    }
}

File: ohheck.help/src/types/admin/notification.ts
import { ModelBase } from '../commontypes/modelbase';

export class Notification extends ModelBase {
    level: string;
    text: string;
    seen: boolean;
    action?: {
        type: string;
        location: string;
        text: string;
    }

    constructor(json) {
        super(json);

        this.level = json.level;
        this.text = json.text;
        this.seen = json.seen;

        if (json.action) {
            this.action = {
                type: json.action.type,
                location: json.action.location,
                text: json.action.text
            };
        }
    }
}